buildmodel/buildassets/buildassets.go

```diff
@@ -148,16 +162,21 @@ func (b BuildResultsDirectoryInfo) CreateSummary() (*BuildAssets, error) {
 
 // getVersion reads the file at path, if it exists. If it doesn't exist, returns the default
 // provided by the caller. If the file cannot be read for some other reason, panics. This logic
-// helps with the "VERSION" files that are only present in Go release branches.
+// helps with the "VERSION" files that are only present in Go release branches, and handles unusual
+// VERSION files that may contain a newline by only reading the first line.
 func getVersion(path string, defaultVersion string) (version string) {
-	bytes, err := ioutil.ReadFile(path)
+	f, err := os.Open(path)
 	if err != nil {
 		if errors.Is(err, os.ErrNotExist) {
 			return defaultVersion
 		}
 		log.Panic(err)
 	}
-	return string(bytes)
+	defer f.Close()
+
+	s := bufio.NewScanner(f)
+	s.Scan()
```

Comment:
`bufio.Scan` returns false when the scan stops, either by reaching the end of the input or an error. If you are confident there is no need to handle the error, the idiomatic way to do so is `_ = s.Scan()`.

Nit: Could this function return an error instead of panicking? It is preferable to avoid panics inside non-main packages so they are easy to test and reuse.

Reply:
`s.Scan` returns a `bool` (whether the scan stopped), not an `error`, so maybe that makes it unnecessary? My IDE warns me about missing error handling, but doesn't warn me about this case. In any case, I'll add it to make it explicit.

I do need to make sure to check `s.Err()` though.

I'll change this to return an error. Some earlier code I wrote had more panics, but I've been moving away from that. (Some of the conversion into `err` I've done so far has been driven by wanting to have the fork sync code finish as many updates as possible, even if an error happens in some particular branch.) Just haven't gotten around to changing this case.